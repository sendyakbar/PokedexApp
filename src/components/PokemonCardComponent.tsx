import React from 'react';
import {
  StyleSheet,
  Image,
  Text,
  ViewStyle,
  TouchableOpacity,
  ImageStyle,
  TextStyle,
} from 'react-native';

type Props = {
  data: {
    name: string;
    url: string;
  };
  onPress: () => void;
};

export default function PokemonCardComponent(props: Props) {
  const {data, onPress} = props;

  return (
    <TouchableOpacity
      style={styles.container}
      activeOpacity={0.8}
      onPress={onPress}>
      <Image
        source={require('../assets/images/pokeball.png')}
        style={styles.pokeball}
      />
      <Text style={styles.itemNameText}>{data.name}</Text>
    </TouchableOpacity>
  );
}

type StyleType = {
  container: ViewStyle;
  pokeball: ImageStyle;
  itemNameText: TextStyle;
};

const styles = StyleSheet.create<StyleType>({
  container: {
    width: '49%',
    backgroundColor: 'white',
    borderRadius: 4,
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
    elevation: 3,
    shadowColor: '#000000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.17,
    shadowRadius: 2.54,
  },
  pokeball: {
    height: 36,
    width: 36,
  },
  itemNameText: {
    marginLeft: 8,
    fontSize: 14,
    color: 'black',
    fontWeight: '400',
  },
});
