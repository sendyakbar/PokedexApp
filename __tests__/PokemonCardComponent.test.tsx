import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import PokemonCardComponent from '../src/components/PokemonCardComponent';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <PokemonCardComponent
        data={{name: 'Pokemon', url: 'https://pokemon.com'}}
        onPress={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
