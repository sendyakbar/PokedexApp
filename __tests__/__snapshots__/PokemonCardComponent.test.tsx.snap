// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 4,
      "elevation": 3,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 8,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2.54,
      "width": "49%",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/pokeball.png",
      }
    }
    style={
      {
        "height": 36,
        "width": 36,
      }
    }
  />
  <Text
    style={
      {
        "color": "black",
        "fontSize": 14,
        "fontWeight": "500",
        "marginLeft": 8,
      }
    }
  >
    Pokemon
  </Text>
</View>
`;
